src/s_inter.c

```diff
@@ -1703,3 +2403,39 @@ void pd_globallock(void) {}
 void pd_globalunlock(void) {}
 
 #endif /* PDTHREADS */
+
+#ifdef THREADED_IO
+static pthread_mutex_t sys_mutexio = PTHREAD_MUTEX_INITIALIZER;
+
+void sys_lockio()
+{
+    int ret = pthread_mutex_lock(&sys_mutexio);
+    if(ret)
+        printf("lockio: %d\n", ret);
+}
+
+void sys_unlockio()
+{
+    int ret = pthread_mutex_unlock(&sys_mutexio);
+    if(ret)
+        printf("unlockio: %d\n", ret);
+}
+#else // THREADED_IO
+#ifdef TEST_LOCKING /* run standalone Pd with this to find deadlocks */
+static int amlockedio;
+void sys_lockio(void)
+{
+    if (amlockedio) bug("duplicate lockio");
+    amlockedio = 1;
+}
+
+void sys_unlock(void)
```

Comment:
should be `sys_unlockio` I guess

Reply:
sure enough! 861697903993e876d08ef487b137994c8681614e